plugin/main/src/kotlinx/benchmark/gradle/BenchmarksPlugin.kt

```diff
@@ -35,13 +35,9 @@ class BenchmarksPlugin : Plugin<Project> {
             return // TODO: Do we need to fail build at this point or just ignore benchmarks?
         }
 
-        val kotlinClass = tryGetClass<KotlinBasePluginWrapper>("org.jetbrains.kotlin.gradle.plugin.KotlinBasePluginWrapper")
-        if (kotlinClass != null) {
-            plugins.findPlugin(kotlinClass)?.run {
-                logger.info("Detected Kotlin plugin version '${project.getKotlinPluginVersion()}'")
-                if (VersionNumber.parse(project.getKotlinPluginVersion()) < VersionNumber(1, 7, 20, null))
-                    logger.error("JetBrains Gradle Benchmarks plugin requires Kotlin version 1.7.20 or higher")
-            }
+        logger.info("Detected Kotlin plugin version '${project.getKotlinPluginVersion()}'")
```

Comment:
I would advice to approach to Kotlin Plugin dependency in a lazy way:
```kotlin
project.plugins.withType(org.jetbrains.kotlin.gradle.plugin.KotlinBasePlugin::class.java) { kotlinPlugin ->
    ...
    // as well
    logger.info("Detected Kotlin plugin version '${kotlinPlugin.pluginVersion)}'")   
}
```

This also uses KGP-api rather then KGP internal methods.

Reply:
Done